dashboard/app/views/levels/_dialog.html.haml

```diff
@@ -125,20 +125,38 @@
     adjustedScroll = true;
   }
 
+  // TODO(dave): Dashboard shouldn't be reaching into the internal implementation of
+  // individual levels. Instead levels should call appOptions.onAttempt.
   $('.submitButton').click(function () {
     var submitButton = $('.submitButton');
     if (submitButton.attr('disabled')) {
       return;
     }
 
+    // Avoid multiple simultaneous submissions.
+    submitButton.attr('disabled', true);
+
+    var onCompleteNoRedirect = function() {
+      $('.submitButton').attr('disabled', false);
+    };
+
+    processResults(onCompleteNoRedirect);
+  });
+
+  /**
+   * Process the solution to the puzzle submitted by the user.
+   * @param {Function} onCompleteNoRedirect Optional callback function to call if
```

Comment:
It does seem weird to offer such a specific callback. I've changed this to be onComplete, but since pixelation shouldn't know about the implementation details of processResults (including lastServerResponse), it just receives a boolean `willRedirect`.